genesis/utils/usd.py

```diff
@@ -0,0 +1,312 @@
+            # check surface
+            mesh_surface = gs.surfaces.Default() if surface is None else surface.copy()
+
+            # load mesh basic information
+            mesh = UsdGeom.Mesh(prim)
+            points = mesh.GetPointsAttr().Get()
+            normals = mesh.GetNormalsAttr().Get()
+            face_vertex_counts = mesh.GetFaceVertexCountsAttr().Get()
+            face_vertex_indices = mesh.GetFaceVertexIndicesAttr().Get()
+            sts = mesh.GetPrim().GetAttribute("primvars:st").Get()
+            st_indices = mesh.GetPrim().GetAttribute("primvars:st:indices").Get()
+
+            # calculate transform
+            transform = np.array(get_world_transform(mesh.GetPrim()))
+            points_homogeneous = np.hstack([points, np.ones((len(points), 1))])
+
+            verts = (points_homogeneous @ transform)[:, :3]
+            normals = np.array(normals, dtype=float)
+            indices = np.array(face_vertex_indices, dtype=int)
+
+            faces = []
+            index_offset = 0
+            face_count = {}
+
+            for count in face_vertex_counts:
+                face_count[count] = face_count.get(count, 0) + 1
+                face = indices[index_offset : index_offset + count].tolist()
+
+                # If the face is a triangle, keep it as is
+                if count == 3:
+                    faces.append(face)
+                # If the face is a quad or more, triangulate it
+                elif count > 3:
+                    for i in range(1, count - 1):
+                        faces.append([face[0], face[i], face[i + 1]])  # Fan triangulation
+
+                index_offset += count  # Move to the next set of indices
+
+            # if face count is not 3, means it is not all triangle faces
+            if len(face_count) >= 2 or (3 not in face_count):
+                gs.logger.warning(f"Mesh {mesh} has non-triangle faces. Face count {face_count}")
+
+            faces = np.array(faces, dtype=int)
+
+            # load uvs
+            if sts is not None:
+                uvs = []
+                gs.logger.info("Parsing OpenUSD Face st into Vertex uv")
+                # Create a mapping of face vertex indices to their UV
+                if st_indices:
+                    assert len(face_vertex_indices) == len(
+                        st_indices
+                    ), "face vertice count should be the same as st indices"
+                    index_to_uv = {fv_idx: sts[st_indices[idx]] for idx, fv_idx in enumerate(face_vertex_indices)}
+                else:
+                    assert len(sts) == len(face_vertex_indices), "face vertice count should be the same as st"
+                    index_to_uv = {fv_idx: sts[idx] for idx, fv_idx in enumerate(face_vertex_indices)}
+
+                # Collect the UVs using the mapping
+                uvs = [index_to_uv[i] for i in range(len(points)) if i in index_to_uv]
```

Comment:
Are you sure about this condition `if i in index_to_uv` ? it means having less uv values than points is supported.